mmdet3d/datasets/det3d_dataset.py

```diff
@@ -318,12 +326,34 @@ def prepare_data(self, index: int) -> Optional[dict]:
                 return None
 
         example = self.pipeline(input_dict)
+
         if not self.test_mode and self.filter_empty_gt:
             # after pipeline drop the example with empty annotations
             # return None to random another in `__getitem__`
             if example is None or len(
                     example['data_samples'].gt_instances_3d.labels_3d) == 0:
                 return None
+
+        if self.show_ins_statistics:
```

Comment:
1. Wrap it into a callable function.
2. We can simply do it once after the data is loaded. 
3. Show the information dataset length, class names, and number of instances of each class